src/System.IO.FileSystem.Watcher/src/System/IO/PatternMatcher.cs

```diff
@@ -227,7 +227,6 @@ public static bool StrictMatchPattern(string expression, string name)
                 if (nameOffset < name.Length)
                 {
                     nameChar = name[nameOffset];
-                    length = 1;
```

Comment:
cc: @ianhays 
This value is never used.  We overwrite the local before we look at it.